pkg/assets/store.go

```diff
@@ -113,6 +138,29 @@ func (s *StoreBuilder) AddProxyConfig(ctx context.Context, namespace string, pc
 	return nil
 }
 
+// AddCustomHTTPConfig processes the given *CustomHTTPConfig and adds the referenced credentials to the store.
+func (s *StoreBuilder) AddCustomHTTPConfig(ctx context.Context, ns string, pc monitoringv1.CustomHTTPConfig) error {
+	if err := pc.Validate(); err != nil {
+		return err
+	}
+
+	for _, header := range pc.HTTPHeaders {
+		// Make sure there are no reference reserved headers
+		if _, ok := reservedHeaders[http.CanonicalHeaderKey(header.Name)]; ok {
```

Comment:
the validation shouldn't be done by StoreBuilder IMHO.

Reply:
Thanks @simonpasquier 
Got it.  Move validation to apis package